orttraining/orttraining/models/bert/main.cc

```diff
@@ -504,13 +504,22 @@ const std::map<std::string, std::pair<std::string, size_t>> input_to_dimension_m
 MapStringToString mapped_dimensions;
 
 void setup_training_params(BertParameters& params) {
-  params.model_path = ToPathString(params.model_name) + ORT_TSTR(".onnx");
-  params.model_with_loss_func_path = ToPathString(params.model_name) + ORT_TSTR("_with_cost.onnx");
-  params.model_with_training_graph_path = ToPathString(params.model_name) + ORT_TSTR("_bw.onnx");
-  params.model_actual_running_graph_path = ToPathString(params.model_name) + ORT_TSTR("_bw_running.onnx");
+  auto model_name_base = ToPathString(params.model_name);
+  params.model_path = model_name_base + ORT_TSTR(".onnx");
+  params.model_with_loss_func_path = model_name_base + ORT_TSTR("_with_cost.onnx");
+  params.model_with_training_graph_path = model_name_base + ORT_TSTR("_bw.onnx");
+  params.model_actual_running_graph_path = model_name_base + ORT_TSTR("_bw_running.onnx");
 
 #ifdef USE_HOROVOD
   params.mpi_context = setup_horovod();
+
+  if (params.pipeline_parallel_size > 1) {
+    auto pipeline_model_name_base = model_name_base + ToPathString(std::to_string(params.mpi_context.world_rank));
```

Comment:
>params.mpi_context.world_rank [](start = 82, length = 29)

again, this is hardcoded assumption for D/H/P config

Reply:
The check for DP v.s. MP is on line 516 -- We only surfix the file name if pipeline is used. 

---
In reply to: [434895616](https://github.com/microsoft/onnxruntime/pull/4113#discussion_r434895616) [](ancestors = 434895616)

Reply:
What I mean is that the name corresponds to pipeline stage, which is hardcoded to world_rank here. Ideally we should have a util function to map from mpi world rank to pipeline stage.

---
In reply to: [437010483](https://github.com/microsoft/onnxruntime/pull/4113#discussion_r437010483) [](ancestors = 437010483,434895616)